### PR TITLE
Core/Scenario - Fix outdated ScenarioState packet structure

### PR DESCRIPTION
**Pull request by Krudor**
**Changes proposed:**

-  Fix Scenario State packet by updating the structure to include a new missing property.

**Issues addressed:**

-  No open issues found.

**Tests performed:**

- Builds
- Play-testing
  - Scenario - Stormwind Stockades (510)
    - Solo testing "Kill X, Y and Z" for scenario with a single step.
    - Packet is now broadcast to client with expected results.

**Known issues and TODO list:** (add/remove lines as needed)

- New packet property is not assigned a real value.

### CODE REVIEW

File: src/server/game/Server/Packets/ScenarioPackets.cpp [hunk 1]
```diff
@@ -46,6 +46,7 @@ WorldPacket const* WorldPackets::Scenario::ScenarioState::Write()
     _worldPacket << uint32(BonusObjectives.size());
     _worldPacket << uint32(PickedSteps.size());
     _worldPacket << uint32(Spells.size());
+    _worldPacket << PlayerGuidUnknown;
```
Comment (MaxtorCoder) [inline src/server/game/Server/Packets/ScenarioPackets.cpp:49]:
```suggestion
    _worldPacket << PlayerGUID;
```

File: src/server/game/Server/Packets/ScenarioPackets.h [hunk 2]
```diff
@@ -59,6 +59,7 @@ namespace WorldPackets
             std::vector<BonusObjectiveData> BonusObjectives;
             std::vector<uint32> PickedSteps;
             std::vector<ScenarioSpellUpdate> Spells;
+            ObjectGuid PlayerGuidUnknown = ObjectGuid::Empty;
```
Comment (MaxtorCoder) [inline src/server/game/Server/Packets/ScenarioPackets.h:62]:
```suggestion
            ObjectGuid PlayerGUID = ObjectGuid::Empty;
```

**Review by MaxtorCoder — CHANGES_REQUESTED** [review]
Field name is just `PlayerGUID`

**Comment by Krudor** [comment]
Noticed from sniffs where scenario was run by a single player that it's just that player's guid.
Still don't know it's purpose and what will happen with multiple players, if it's the initiator/owner of the scenario or what.

Might do some investigations in the future with multiple characters and see if it ever changes, or if it does, what it might be indicative of.

**Pull request merged**